core/coreunix/add.go

```diff
@@ -173,48 +173,63 @@ func (adder *Adder) PinRoot() error {
 	return adder.node.Pinning.Flush()
 }
 
-func (adder *Adder) Finalize() (*dag.Node, error) {
+func (adder *Adder) Finalize(write bool) (*dag.Node, error) {
+	root := adder.mr.GetValue()
+
 	// cant just call adder.RootNode() here as we need the name for printing
-	root, err := adder.mr.GetValue().GetNode()
+	rootNode, err := root.GetNode()
 	if err != nil {
 		return nil, err
 	}
 
 	var name string
 	if !adder.Wrap {
-		name = root.Links[0].Name
-		child, err := root.Links[0].GetNode(adder.ctx, adder.node.DAG)
+		name = rootNode.Links[0].Name
+
+		root, err = adder.mr.GetValue().(*mfs.Directory).Child(name)
 		if err != nil {
 			return nil, err
 		}
-		root = child
 	}
 
 	err = adder.outputDirs(name, root)
 	if err != nil {
 		return nil, err
 	}
 
-	err = adder.mr.Close()
+	if write {
```

Comment:
no. We want to ensure (from a top level perspective) that the mfs structure is properly cleaned up, it triggers an update if something registered for updates, but this instance of it doesnt.